cpp/include/raft/sparse/neighbors/knn.cuh

```diff
@@ -103,4 +106,171 @@ void brute_force_knn(const value_idx* idxIndptr,
+                                       queryData.data(),
+                                       queryIndices.size(),
+                                       queryIndptr.size() - 1,
+                                       csr_query.structure_view().get_n_cols(),
+                                       output_indices.data_handle(),
+                                       output_dists.data_handle(),
+                                       k,
+                                       handle,
+                                       batch_size_index,
+                                       batch_size_query,
+                                       metric,
+                                       metricArg);
+}
+
+/**
+ * Search the sparse kNN for the k-nearest neighbors of a set of sparse query vectors
+ * using some distance implementation
+ * @param[in] coo_idx index coo matrix
+ * @param[in] coo_query query coo matrix
+ * @param[out] output_indices dense matrix for output indices (size n_query_rows * k)
+ * @param[out] output_dists dense matrix for output distances (size n_query_rows * k)
+ * @param[in] k the number of neighbors to query
+ * @param[in] handle CUDA resource::get_cuda_stream(handle) to order operations with respect to
+ * @param[in] batch_size_index maximum number of rows to use from index matrix per batch
+ * @param[in] batch_size_query maximum number of rows to use from query matrix per batch
+ * @param[in] metric distance metric/measure to use
+ * @param[in] metricArg potential argument for metric (currently unused)
+ */
+template <typename value_idx = int, typename value_t = float, int TPB_X = 32>
+void brute_force_knn(raft::device_coo_matrix<value_t,
+                                             value_idx,
+                                             value_idx,
+                                             value_idx,
+                                             raft::device_uvector_policy,
+                                             raft::PRESERVING> coo_idx,
+                     raft::device_coo_matrix<value_t,
+                                             value_idx,
+                                             value_idx,
+                                             value_idx,
+                                             raft::device_uvector_policy,
+                                             raft::PRESERVING> coo_query,
+                     device_vector_view<value_idx> output_indices,
+                     device_vector_view<value_t> output_dists,
+                     int k,
+                     raft::resources const& handle,
+                     size_t batch_size_index             = 2 << 14,  // approx 1M
+                     size_t batch_size_query             = 2 << 14,
+                     raft::distance::DistanceType metric = raft::distance::DistanceType::L2Expanded,
+                     float metricArg                     = 0)
+{
+  cudaStream_t stream = raft::resource::get_cuda_stream(handle);
+
```

Comment:
Maybe we could add a judgment for `0` size data for `idx` and `query`, though it should happen rarely.

Reply:
@rhdong do you think I should raise and error or just return before performing bfknn?

Reply:
Should depend on the logic:  to return directly (keeping no change on the outputs), if it is normal to have zero-size input, or you could use `RAFT_EXPECTS` to notify the caller.